test/e2e/common/node/framework/cgroups/cgroups.go

```diff
@@ -0,0 +1,421 @@
+/*
+Copyright 2024 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cgroups
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+	"strings"
+
+	"github.com/onsi/ginkgo/v2"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	utilerrors "k8s.io/apimachinery/pkg/util/errors"
+	kubecm "k8s.io/kubernetes/pkg/kubelet/cm"
+	"k8s.io/kubernetes/test/e2e/framework"
+	e2epod "k8s.io/kubernetes/test/e2e/framework/pod"
+	imageutils "k8s.io/kubernetes/test/utils/image"
+
+	"github.com/onsi/gomega"
+)
+
+const (
+	cgroupFsPath          string = "/sys/fs/cgroup"
+	cgroupCPUSharesFile   string = "cpu.shares"
+	cgroupCPUQuotaFile    string = "cpu.cfs_quota_us"
+	cgroupMemLimitFile    string = "memory.limit_in_bytes"
+	cgroupv2CPUWeightFile string = "cpu.weight"
+	cgroupv2CPULimitFile  string = "cpu.max"
+	cgroupv2MemLimitFile  string = "memory.max"
+	cgroupVolumeName      string = "sysfscgroup"
+	cgroupMountPath       string = "/sysfscgroup"
+)
+
+var (
+	podOnCgroupv2Node *bool
```

Comment:
I don't think we run any e2e suites on mixed cgroups v1 and v2 clusters, but technically this shouldn't really be cached as a global. Maybe just add a note or TODO here?

Reply:
That's right. I added a TODO.